### PR TITLE
BUG: Fix Dockerfile apt-get installs 

### PR DESCRIPTION
**Pull request by trallard**
#### What does this implement/fix?

- Fixes the apt-get installs that was failing
- Removes duplicated deps from the `.gitpod.Dockerfile`

### CODE REVIEW

**Comment by rlucas7** [comment]
Yeah, looks like my reorder dropped the ampersands: 
https://github.com/scipy/scipy/pull/13551/files#diff-b96c73c473cf5639a7f375cc6fda5dbc149648c1e90cca69b21cd0ea6b5f43bfL86
thanks @trallard

File: tools/docker_dev/.gitpod.Dockerfile [hunk 1]
```diff
@@ -10,8 +10,6 @@ FROM ${BASE_CONTAINER}
 RUN apt-get update && \ 
     apt-get install -yq --no-install-recommends \
     bash-completion \
-    build-essential \
-    git \
```
Comment (rlucas7) [inline tools/docker_dev/.gitpod.Dockerfile:14]: ah I see now the the env dependency relnship here between the gitpod and the docker

**Review by rlucas7 — APPROVED** [review]
Marking approved, I'll leave for @rgommers to merge though, I don't think I have permissions on the docker side if this breaks again to investigate. I did link to what appears to be the offending line in my reorder so should be straightforward. Thanks again.

**Review by rgommers — APPROVED** [review]
Thanks Tania!

**Pull request merged**